test/SafeCallback.t.sol

```diff
@@ -0,0 +1,29 @@
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import {Deployers} from "@uniswap/v4-core/test/utils/Deployers.sol";
+
+import {SafeCallback} from "../src/base/SafeCallback.sol";
+import {MockSafeCallback} from "./mocks/MockSafeCallback.sol";
+
+contract SafeCallbackTest is Test, Deployers {
+    MockSafeCallback safeCallback;
+
+    function setUp() public {
+        deployFreshManager();
+        safeCallback = new MockSafeCallback(manager);
+    }
+
+    function test_unlock(uint256 num) public {
```

Comment:
can we also just add an assert that the pool manager in SafeCallback is the address we expect/the pool manager address we initialized it with